### PR TITLE
[ECO-2354] Add slippage and gas on trading page

### PR DESCRIPTION
**Pull request by CRBl69**
# Description

This PR adds the gear for settings and the gas display on the trading page.

## Screenshots

image

image

image

# Testing

See vercel.

### CODE REVIEW

File: src/typescript/frontend/src/components/pages/emojicoin/components/trade-emojicoin/SwapButton.tsx
Comment (alnoki) [inline src/typescript/frontend/src/components/pages/emojicoin/components/trade-emojicoin/SwapButton.tsx:1]: - [ ] Use `APT` in the gas amount printout
- [ ] Hard-code the amount for when using Aptos Connect

**Review by xbtmatt — CHANGES_REQUESTED** [review]
It looks like you have an infinite re-render issue somewhere, when I look at the preview page I get this:

```shell
Maximum update depth exceeded. This can happen when a component 
repeatedly calls setState inside componentWillUpdate or 
componentDidUpdate. React limits the number of nested updates to prevent
infinite loops.
```

**Comment by CRBl69** [comment]
@xbtmatt what page are you getting that on exactly ? I can't reproduce it.

**Comment by CRBl69** [comment]
Nevermind, I fixed it.

File: src/typescript/frontend/src/components/selects/trade-options/index.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                  setMaxSlippageMode("auto");
+                  SlippageSettings.setMaxSlippageMode("auto");
+                  setMaxSlippage(SlippageSettings.getMaxSlippageSettings().maxSlippage);
+                  if (onMaxSlippageUpdate) onMaxSlippageUpdate();
+                }}
+              >
+                AUTO
+              </span>
+              <span
+                className={`${maxSlippageMode === "custom" ? "text-black" : "text-dark-gray"} font-pixelar text-2xl cursor-pointer`}
```
Comment (xbtmatt) [inline src/typescript/frontend/src/components/selects/trade-options/index.tsx:63]:
```suggestion
                className={`${maxSlippageMode === "custom" ? "opacity-30" : "opacity-100"} text-black font-pixelar text-2xl cursor-pointer`}
```

File: src/typescript/frontend/src/components/selects/trade-options/index.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                  setMaxSlippageMode("custom");
+                  SlippageSettings.setMaxSlippageMode("custom");
+                  if (onMaxSlippageUpdate) onMaxSlippageUpdate();
+                }}
+              >
+                CUSTOM
+              </span>
+            </div>
+            <div
+              className={`${maxSlippageMode === "custom" ? "text-black border-black" : "text-dark-gray border-dark-gray"} flex flex-row border-[1px] p-[.2rem] rounded border-solid`}
```
Comment (xbtmatt) [inline src/typescript/frontend/src/components/selects/trade-options/index.tsx:74]:
```suggestion
              className={`${maxSlippageMode === "custom" ? "opacity-100" : "opacity-30"} text-black border-black flex flex-row border-[1px] p-[.2rem] rounded border-solid`}
```

File: src/typescript/frontend/src/components/selects/trade-options/index.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            <span className="text-black font-pixelar text-2xl">{Number(maxSlippage) / 100}%</span>
+          </div>
+        </DropdownMenuInner>
+      </StyledDropdownMenuItem>
+      <StyledDropdownMenuItem disabled={false}>
+        <DropdownMenuInner>
+          <div className="med-pixel-text flex flex-row justify-between gap-[32px]">
+            <div className="mt-[3px] med-pixel-text flex flex-row justify-between gap-[16px]">
+              <span
+                className={`${maxSlippageMode === "auto" ? "text-black" : "text-dark-gray"} font-pixelar text-2xl cursor-pointer`}
```
Comment (xbtmatt) [inline src/typescript/frontend/src/components/selects/trade-options/index.tsx:52]:
```suggestion
                className={`${maxSlippageMode === "auto" ? "opacity-100" : "opacity-30"} text-black font-pixelar text-2xl cursor-pointer`}
```

File: src/typescript/frontend/src/components/selects/trade-options/index.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import React, { useEffect, useState } from "react";
+
+import { DropdownMenuWrapper } from "../dropdown-menu/styled";
+
+import { type TradeOptionsProps } from "../types";
+import {
+  DropdownMenuInner,
+  StyledDropdownMenuClose,
```
Comment (xbtmatt) [inline src/typescript/frontend/src/components/selects/trade-options/index.tsx:8]:
```suggestion
```

File: src/typescript/frontend/src/components/selects/trade-options/index.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        onClick={onClose}
+        className="cursor-pointer"
+      >
+        <DropdownMenuInner
+          className="flex"
+          style={{ justifyContent: "end", margin: "0", padding: "0" }}
+        >
+          <div className="med-pixel-text lowercase">x</div>
+        </DropdownMenuInner>
+      </StyledDropdownMenuClose>
```
Comment (xbtmatt) [inline src/typescript/frontend/src/components/selects/trade-options/index.tsx:38]:
```suggestion
```

Much cleaner without the close here imo, plus it already disappears if you click off of it, like the wallet adapter dropdown. Looks incredible this way too :)

File: src/typescript/frontend/src/lib/hooks/queries/use-simulate-swap.ts [hunk 6]
```diff
@@ -12,22 +12,58 @@ import { withResponseError } from "./client";
...
+        swapperPubKey: publicKey,
+        marketAddress: args.marketAddress,
+        inputAmount: args.inputAmount,
+        isSell: args.isSell,
+        integrator: INTEGRATOR_ADDRESS,
+        integratorFeeRateBPs: INTEGRATOR_FEE_RATE_BPS,
+        minOutputAmount: args.minOutputAmount,
+        typeTags: args.typeTags,
+      });
+      const swapName = `${MODULE_ADDRESS}::${EMOJICOIN_DOT_FUN_MODULE_NAME}::Swap`;
```
Comment (xbtmatt) [inline src/typescript/frontend/src/lib/hooks/queries/use-simulate-swap.ts:44]:
```suggestion
```

File: src/typescript/frontend/src/lib/hooks/queries/use-simulate-swap.ts [hunk 7]
```diff
@@ -12,22 +12,58 @@ import { withResponseError } from "./client";
 import Big from "big.js";
 import { useMemo } from "react";
 import { toCoinTypes } from "@sdk/markets/utils";
+import { type AccountInfo } from "@aptos-labs/wallet-adapter-core";
+import { tryEd25519PublicKey } from "components/pages/launch-emojicoin/hooks/use-register-market";
+import { EMOJICOIN_DOT_FUN_MODULE_NAME, MODULE_ADDRESS } from "@sdk/const";
```
Comment (xbtmatt) [inline src/typescript/frontend/src/lib/hooks/queries/use-simulate-swap.ts:17]:
```suggestion
import { STRUCT_STRINGS } from "@sdk/utils";
```

File: src/typescript/frontend/src/lib/hooks/queries/use-simulate-swap.ts [hunk 8]
```diff
@@ -12,22 +12,58 @@ import { withResponseError } from "./client";
...
+        marketAddress: args.marketAddress,
+        inputAmount: args.inputAmount,
+        isSell: args.isSell,
+        integrator: INTEGRATOR_ADDRESS,
+        integratorFeeRateBPs: INTEGRATOR_FEE_RATE_BPS,
+        minOutputAmount: args.minOutputAmount,
+        typeTags: args.typeTags,
+      });
+      const swapName = `${MODULE_ADDRESS}::${EMOJICOIN_DOT_FUN_MODULE_NAME}::Swap`;
+      const swapEvent = res.events.find((e) => e.type === swapName)!;
```
Comment (xbtmatt) [inline src/typescript/frontend/src/lib/hooks/queries/use-simulate-swap.ts:45]:
```suggestion
      const swapEvent = res.events.find((e) => e.type === STRUCT_STRINGS.SwapEvent)!;
```

**Review by xbtmatt — APPROVED** [review]
This looks incredible! Thanks for doing this, it's really slick, love the way it looks.

File: src/typescript/frontend/src/components/pages/emojicoin/components/trade-emojicoin/SwapButton.tsx
Comment (xbtmatt) [reply src/typescript/frontend/src/components/pages/emojicoin/components/trade-emojicoin/SwapButton.tsx:1]: Resolving this as both were addressed!

File: src/typescript/frontend/src/components/selects/trade-options/index.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                  setMaxSlippageMode("auto");
+                  SlippageSettings.setMaxSlippageMode("auto");
+                  setMaxSlippage(SlippageSettings.getMaxSlippageSettings().maxSlippage);
+                  if (onMaxSlippageUpdate) onMaxSlippageUpdate();
+                }}
+              >
+                AUTO
+              </span>
+              <span
+                className={`${maxSlippageMode === "custom" ? "opacity-30" : "opacity-100"} text-black font-pixelar text-2xl cursor-pointer`}
```
Comment (xbtmatt) [inline src/typescript/frontend/src/components/selects/trade-options/index.tsx:49]:
```suggestion
                className={`${maxSlippageMode === "custom" ? "opacity-100" : "opacity-30"} text-black font-pixelar text-2xl cursor-pointer`}
```

**Pull request merged**